src/mscorlib/shared/System/Collections/Generic/Dictionary.cs

```diff
@@ -508,21 +530,34 @@ private void Resize()
             Resize(HashHelpers.ExpandPrime(_count), false);
         }
 
-        private void Resize(int newSize, bool forceNewHashCodes)
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        private void Rehash()
```

Comment:
I am not sure whether separating this into a non-inlineable is helping in either JIT or AOT case. There is a fixed memory overhead per generic method even when it is not JITed (ie the cost of the generics is more than the cost of the code). I would keep it where it was.